Paralayout/UIView+Alignment.swift

```diff
@@ -115,6 +115,80 @@ extension Alignable {
+    ///
+    /// - parameter position: The position in the receiving view's untransformed frame.
+    /// - parameter otherView: The other view for the measurement.
+    /// - parameter otherPosition: The position in the `otherView`'s untransformed frame to use for the measurement.
+    /// - returns: The offset from the receiver's `position` to the `otherView`'s `otherPosition`.
+    public func untransformedFrameOrigin(
+        toAlign position: Position,
+        to otherView: Alignable,
+        _ otherPosition: Position,
+        alignmentBehavior: TargetAlignmentBehavior = .automatic
+    ) throws -> CGPoint {
+        let receiverContext = alignmentContext
+        let receiverView = receiverContext.view
+
+        let targetContext = otherView.alignmentContext
+        let targetView = targetContext.view
+
+        // We can't be aligned to another view if we don't have a superview.
+        guard let superview = receiverView.superview else {
+            ParalayoutAlertForInvalidViewHierarchy()
+            return .zero
+        }
+
+        switch position {
+        case .topLeft, .topRight, .leftCenter, .rightCenter, .bottomLeft, .bottomRight:
+            switch otherPosition {
+            case .topLeading, .topTrailing, .leadingCenter, .trailingCenter, .bottomLeading, .bottomTrailing:
+                ParalayoutAlertForMismatchedAlignmentPositionTypes()
+            default:
+                break
+            }
+
+        case .topLeading, .topTrailing, .leadingCenter, .trailingCenter, .bottomLeading, .bottomTrailing:
+            switch otherPosition {
+            case .topLeft, .topRight, .leftCenter, .rightCenter, .bottomLeft, .bottomRight:
+                ParalayoutAlertForMismatchedAlignmentPositionTypes()
+            default:
+                break
+            }
+
+        default:
+            break
+        }
+
+        let targetIsInSourceSuperviewChain = sequence(first: receiverView, next: { $0.superview }).contains(targetView)
+
+        let targetPoint: CGPoint
+        switch alignmentBehavior {
+        case .bounds,
+                .automatic where targetIsInSourceSuperviewChain:
+            targetPoint = try superview.untransformedConvert(
+                targetContext
+                    .pointInBounds(at: otherPosition)
+                    .offset(
+                        by: UIOffset(horizontal: -targetView.bounds.origin.x, vertical: -targetView.bounds.origin.y)
+                    ),
+                from: targetView
+            )
+
+        case .untransformedFrame,
+                .automatic /* where !targetIsInSourceSuperviewChain */:
+            targetPoint = try superview.untransformedConvert(
+                targetContext.pointInBounds(at: otherPosition),
+                from: targetView
+            )
+        }
+
+        let sourcePoint = receiverContext.pointInBounds(at: position)
+        return targetPoint.offset(by: .init(horizontal: -sourcePoint.x, vertical: -sourcePoint.y))
```

Comment:
So I _think_ this is correct for the base case of calling `superview.untransformedConvert`. Are there snapshot tests that work on views with transformations applied?

Reply:
You know what I know there. Need to defer to @NickEntin

Reply:
these last two lines are the only difference between the above method and this one, right?